strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/archive/Bzip2ArchiveListingFunction.java

```diff
@@ -0,0 +1,44 @@
+package org.carlspring.strongbox.artifact.archive;
+
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Set;
+
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public enum Bzip2ArchiveListingFunction
+        implements ArchiveListingFunction
+{
+
+    INSTANCE;
+
+    @Override
+    public Set<String> listFilenames(final RepositoryPath path)
```

Comment:
I'd say it hsould be called `listFileNames` (with a capital 'N')...

Reply:
Hmm how about this https://docs.oracle.com/javase/7/docs/api/java/io/FilenameFilter.html ? Do you really want capital 'N' ?

Reply:
Ah... I see... Okay.